sdk/src/azure/iot/az_iot_common.c

```diff
@@ -27,6 +27,7 @@ AZ_NODISCARD az_result az_iot_message_properties_init(
   _az_PRECONDITION_NOT_NULL(properties);
   _az_PRECONDITION_VALID_SPAN(buffer, 0, true);
   _az_PRECONDITION(written_length >= 0);
```

Comment:
nit, you could combine the condition like this:
```suggestion
  _az_PRECONDITION_RANGE(0, written_length, az_span_size(buffer));
```

Reply:
Yea I like this one